src/Reflection/ReflectionClass.php

```diff
@@ -1014,10 +1016,22 @@ function (Node\Name $interfaceName) {
             : [];
     }
 
+
+    private function getInheritanceClassHierarchy()
+    {
+        if ($this->hierarchy) {
+            return $this->hierarchy;
```

Comment:
Generally not, but we have to freeze the assumption that source locators are to be considered immutable, and a memoizing wrapper probably needs to be added there...